scriptworker/worker.py

```diff
@@ -239,6 +239,8 @@ def main(event_loop=None):
         done = True
         if context.running_tasks is not None:
             await context.running_tasks.cancel()
+        # for test coverage
+        sys.exit(1)
```

Comment:
@mitchhentges what do you think about this? This allows us to get full test coverage. Just want to make sure you don't think it'll cause any weirdness.

Reply:
Hey, thanks for pinging me!

I'm not sure that this is a good idea: this signal handler is called when the main process is interrupted (and we don't know where the process was interrupted at). I think it's a lot safer to let the application exit on its own.

I wonder how this `sys.exit(1)` is improving test coverage: is it because it impedes the test runner from exiting normally? :thinking:

Reply:
It's safer, but is it harmful? If it is, I can remove.
It's been a while, but we see the lack of coverage [here](https://travis-ci.org/mozilla-releng/scriptworker/jobs/493004995#L1240). Also, I vaguely remember the test just running forever.

Sounds like I might need to remove this line and see what comes up.

Reply:
It looks like it [works without the `sys.exit`](https://github.com/escapewindow/scriptworker/commit/4d7f838e82a8a3b073fb2b194af52a20da4551c3). We're able to more clearly see which branch we're hitting in the test, because we have slightly different behavior in the `running` branch than if not running. But we can see that we have full coverage, and that we exit the main loop, so perhaps that's sufficient.

Reply:
Weird :thinking: 

FWIW, regarding safer/harmful: I believe that the harm of having the `sys.exit` would be that if/when it affects things negatively, it may be really tricky to determine the cause of the negative behavour. Like, since the main process is interrupted when the signal handler is called, that interruption could happen in the middle of a volatile function (something OS-related? Who knows what the possibilities are). I attribute a high cost to any risks that might cause issues that are incredibly difficult to debug.